infra/cleanup-pipeline.groovy

```diff
@@ -148,7 +149,9 @@ node(params.TARGET_NODE) {
 					images="$(grep -vE '^(infosiftr/moby|oisupport/[^:@]+)(:|$)|^busybox:.+-builder$' <<<"$images")" || :
 
 					# exclude images of running containers if running from an explicit tag
-					containers="$(docker ps --no-trunc --format '{{ .Image }}' | awk '/[:@]/ { print; next } { print $0 ":latest" }')" || return 1
+					# or if it was pulled/run by tag@digest (docker images does not show the tag but docker ps does)
+					containers="$(docker ps --no-trunc --format '{{ .Image }}' | awk '/:[^@]+@/ {str=$0; sub(":[^@]+@","@", str); print str } /[:@]/ { print; next } { print $0 ":latest" }')" || return 1
+
```

Comment:
I tried this and it gave me the same result for both. It's probably something I am doing wrong though.

```bash
> docker run --rm -d alpine@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6 tail -f > /dev/null  

> docker ps --no-trunc --format '{{ .Image }}' | awk '/[:@]/ { print; next } { print $0 ":latest" }' 

alpine@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6
f6648c04cd6c:latest

> docker ps --no-trunc --format '{{ .Image }}' | awk '/:[^@]+@/ {str=$0; sub(":[^@]+@","@", str); print str } /[:@]/ { print; next } { print $0 ":latest" }'

alpine@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6
f6648c04cd6c:latest

```

Reply:
It happens when you pull the image by `tag@digest` (and tag can technically be unrelated to what the digest may be tagged as on the registry).

```console
$ docker run --rm -d --name foo alpine:foo-bar@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6 tail -f /dev/null
$ docker ps
CONTAINER ID   IMAGE            COMMAND               CREATED         STATUS         PORTS     NAMES
de2668cda515   alpine:foo-bar   "tail -f /dev/null"   3 seconds ago   Up 2 seconds             foo
$ docker images alpine
REPOSITORY   TAG       IMAGE ID       CREATED       SIZE
alpine       <none>    f6648c04cd6c   6 weeks ago   7.66MB

$ docker ps --no-trunc --format '{{ .Image }}' | awk '/[:@]/ { print; next } { print $0 ":latest" }'
alpine:foo-bar@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6
$ docker ps --no-trunc --format '{{ .Image }}' | awk '/:[^@]+@/ {str=$0; sub(":[^@]+@","@", str); print str } /[:@]/ { print; next } { print $0 ":latest" }'
alpine@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6
alpine:foo-bar@sha256:b312e4b0e2c665d634602411fcb7c2699ba748c36f59324457bc17de485f36f6
```

Reply:
Ah gotcha!